pkg/rpc/client/rpc.go

```diff
@@ -1010,6 +1019,9 @@ func (c *Client) StateRootInHeader() (bool, error) {
 
 // GetNativeContractHash returns native contract hash by its name.
 func (c *Client) GetNativeContractHash(name string) (util.Uint160, error) {
+	c.cacheLock.Lock() // Cache may be updated inside this method.
+	defer c.cacheLock.Unlock()
+
```

Comment:
We are holding this lock for the whole duration of network request. What about locking/unlocking twice here (if needed)?
For example, we may want to initialize and cache native hashes during application start in parallel. (in theory could also be done with `GetNativeContracts`, but it doesn't update cache).

Reply:
Though, it looks simpler this way, and we can consider updating cache in `GetNativeContracts`.

Reply:
> We are holding this lock for the whole duration of network request.

BTW, it's a _very_ important point for multithreaded networked code.

Reply:
Fixed, now locking is made in two steps where possible.